marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkIngester.sjs

```diff
@@ -29,11 +29,21 @@ const consts = require("/data-hub/5/impl/consts.sjs");
 const HubUtils = require("/data-hub/5/impl/hub-utils.sjs");
 const state = fn.head(xdmp.fromJSON(endpointState));
 
+const temporal = require("/MarkLogic/temporal.xqy");
+const temporalLib = require("/data-hub/5/temporal/hub-temporal.sjs");
```

Comment:
FYI, this library is present only in Datahub 5.3.0. If the connector has to work with all 5.x DataHub, this will fail

Reply:
@srinathgit Is this only needed if the user is not a data-hub-developer (which has the temporal-admin privilege), or does not already have the temporal-admin role? 

The connector can install a copy of hub-temporal.sjs if it doesn't exist - e.g. DHF is prior to 5.3. The amp won't exist, so the function won't do anything. That behavior is consistent with DHF prior to 5.3 - if you don't have the temporal-admin role/privilege, then temporal.collections returns an empty array.